pyogrio/tests/test_geopandas_io.py

```diff
@@ -1214,7 +1234,11 @@ def test_write_None_string_column(tmp_path, use_arrow):
     assert filename.exists()
 
     result_gdf = read_dataframe(filename, use_arrow=use_arrow)
-    assert result_gdf.object_col.dtype == object
+    assert (
```

Comment:
Could this `assert` be folded into the conditional below (with a corresponding `else`)?